src/Bonnier/WP/Cxense/Services/Query.php

```diff
@@ -0,0 +1,224 @@
+<?php
+
+namespace Bonnier\WP\Cxense\Services;
+
+class Query
+{
+    private $widgetId;
+    private $siteId;
+    private $cxenseUserId;
+    private $matchingMode;
+    private $categories;
+    private $tags;
+    private $query;
+    private $context;
+
+    const POPULAR = 'trend';
+    const RELATED = 'contextual'; // Articles similar to the current article.
+    const SIMILAR_READS = 'collabctx'; // People who have read the current article have also read these articles.
+    const RECENTLY_VIEWED = 'recent';
+
+    public function __construct()
+    {
+        $this->setWidgetId(wp_cxense()->settings->get_setting_value('sortby_widget_id', get_locale()));
+        $this->setSiteId(wp_cxense()->settings->get_setting_value("site_id", get_locale()));
+    }
+
+    public static function make()
+    {
+        return new static();
+    }
+
+    /**
+     * @return mixed
+     */
+    public function getContext()
+    {
+        return $this->context;
+    }
+
+    /**
+     * @param mixed $context
+     */
+    public function setContext($context): void
+    {
+        $this->context = $context;
+        $this->query['context'] = $this->getContext();
+    }
+
+    public function addContext(string $key, $value)
+    {
+        if (!isset($this->query['context'])) {
+            $this->query['context'] = [];
+        }
+        $context = $this->getContext();
+        $context[$key] = $value;
+        $this->setContext($context);
+        return $this;
+    }
+
+    /**
+     * @return string
+     */
+    public function getSiteId(): string
+    {
+        return $this->siteId;
+    }
+
+    /**
+     * @param string $siteId
+     */
+    public function setSiteId(string $siteId): void
+    {
+        $this->siteId = $siteId;
+        $this->addParameter('siteId', $this->getSiteId());
+    }
+
+    public function getWidgetId()
+    {
+        return $this->widgetId;
+    }
+
+    /**
+     * @param string $widgetId
+     */
+    private function setWidgetId(string $widgetId): void
+    {
+        $this->widgetId = $widgetId;
+        $this->query['widgetId'] = $this->getWidgetId();
+    }
+
+
+
+    public function addParameter($key, $value)
+    {
+        if (!isset($this->query['parameters'])) {
+            $this->query['parameters'] = [];
+        }
+        array_push($this->query['parameters'], [
+            'key' => $key,
+            'value' => $value
+        ]);
+        return $this;
+    }
+
+    public function bySimilarReads()
+    {
+        $this->setMatchingMode(Query::SIMILAR_READS);
```

Comment:
I think using self or static is better, as you will not have to rename this if class name is changed